### PR TITLE
Add links to entities  in the Notification Architecture Overview doc (#2149)

### LINKED ISSUE #2149: Docs: add links in the Notification Architecture Overview documentation to the entities in the Entities Reference
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

In the [Notification Architecture Overview](https://docs.medusajs.com/advanced/backend/notification/overview) documentation, the entities`NotificationProvider` and `Notification` are mentioned but not linked.

### How can this issue be resolved?

1. Link to each of the entities mentioned earlier under the "Entity Overview" of each. For example, Add a link to the `NotificationProvider` entity under the "NotificationProvider Entity Overview" section. Make sure to only add it to the first time the entity is mentioned in that section.

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

### PR DESCRIPTION
**Pull request by daniloff200**
closes #2149

### CODE REVIEW

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thanks for working on this @daniloff200 but as mentioned in the issue the links should be added to the first time each entity is referenced. So, instead of adding text with the link, add the link to the first time `NotificationProvider` and `Notification` entities are mentioned.

**Comment by daniloff200** [comment]
@shahednasser  hey! Thanks for your comment, I've updated my changes. Can you look at it?

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thanks for making the change, but I meant the first time `NotificationProvider` and `Notification` are mentioned as entities, so it would be under the `NotificationProvider Entity Overview` and `Notification Entity Overview` as well.

**Comment by daniloff200** [comment]
@shahednasser  amother try :) Hope, I understand your requirement correct for this time

File: docs/content/advanced/backend/notification/overview.md [hunk 1]
```diff
@@ -48,7 +47,7 @@ A Notification also represents a resent notification. So, when a notification is
 
 ### Notification Entity Overview
 
-The two most important properties in the `Notification` entity are the `to` and `data` properties. 
+The two most important properties in the [`Notification` entity](../../../references/js-client/classes/internal-11.Notification.md) are the `to` and `data` properties.
```
Comment (shahednasser) [inline docs/content/advanced/backend/notification/overview.md:50]:
```suggestion
The two most important properties in the [`Notification`](../../../references/js-client/classes/internal-11.Notification.md) entity are the `to` and `data` properties. 
```

**Review by shahednasser — COMMENTED** [review]
Yes! just a small change:

**Comment by daniloff200** [comment]
done @shahednasser

**Review by shahednasser — APPROVED** [review]

**Comment by shahednasser** [comment]
Thanks! @daniloff200 🚀

**Pull request merged**